static/app/components/events/eventReplay/replayClipPreview.tsx

```diff
@@ -0,0 +1,333 @@
+                <PlayerScrubber showZoomIndicators />
+              </StyledScrubber>
+              <Time style={{gridArea: 'duration'}}>
+                {duration ? formatTime(duration) : '--:--'}
+              </Time>
+            </TimeAndScrubberGrid>
+          </Container>
+          <ButtonBar gap={1}>
+            <LinkButton size="sm" to={fullReplayUrl}>
+              {t('See Full Replay')}
+            </LinkButton>
+            {showFullscreenButton ? (
+              <Button
+                size="sm"
+                title={isFullscreen ? t('Exit full screen') : t('Enter full screen')}
+                aria-label={isFullscreen ? t('Exit full screen') : t('Enter full screen')}
+                icon={
+                  isFullscreen ? <IconContract size="sm" /> : <IconExpand size="sm" />
+                }
+                onClick={toggleFullscreen}
+                {...fullReplayButtonProps}
+              />
+            ) : null}
+          </ButtonBar>
+        </ButtonGrid>
+      </ErrorBoundary>
+    </Fragment>
+  );
+}
+
+function ReplayClipPreview({
+  eventTimestampMs,
+  orgSlug,
+  replaySlug,
+  fullReplayButtonProps,
+}: Props) {
+  const {fetching, replay, replayRecord, fetchError, replayId} = useReplayReader({
+    orgSlug,
+    replaySlug,
+  });
+  const fullscreenRef = useRef(null);
+  const {toggle: toggleFullscreen} = useFullscreen({
+    elementRef: fullscreenRef,
+  });
+
+  const startTimestampMs = replayRecord?.started_at?.getTime() ?? 0;
+  const eventTimeOffsetMs = Math.abs(eventTimestampMs - startTimestampMs);
+
+  useRouteAnalyticsParams({
+    event_replay_status: getReplayAnalyticsStatus({fetchError, replayRecord}),
+  });
+
+  const clipWindow = useMemo(
+    () => ({
+      startTimeOffsetMs: Math.max(eventTimeOffsetMs - 10e3, 0),
+      durationMs: 15e3,
+    }),
+    [eventTimeOffsetMs]
+  );
+  const offset = useMemo(
+    () => ({offsetMs: clipWindow.startTimeOffsetMs}),
+    [clipWindow.startTimeOffsetMs]
+  );
+
+  if (replayRecord?.is_archived) {
+    return (
+      <Alert type="warning" data-test-id="replay-error">
+        <Flex gap={space(0.5)}>
+          <IconDelete color="gray500" size="sm" />
+          {t('The replay for this event has been deleted.')}
+        </Flex>
+      </Alert>
+    );
+  }
+
+  if (fetchError) {
+    const reasons = [
+      t('The replay is still processing'),
+      tct(
+        'The replay was rate-limited and could not be accepted. [link:View the stats page] for more information.',
+        {
+          link: <Link to={`/organizations/${orgSlug}/stats/?dataCategory=replays`} />,
+        }
+      ),
+      t('The replay has been deleted by a member in your organization.'),
+      t('There were network errors and the replay was not saved.'),
+      tct('[link:Read the docs] to understand why.', {
+        link: (
+          <ExternalLink href="https://docs.sentry.io/platforms/javascript/session-replay/#error-linking" />
+        ),
+      }),
+    ];
```

Comment:
I recently pulled this out into `MissingReplayAlert` which should cut down on the copy+paste

Reply:
Sweet!